core/invoker/src/main/scala/whisk/core/container/ContainerUtils.scala

```diff
@@ -164,14 +164,19 @@ trait ContainerUtils {
             val buffer = java.nio.ByteBuffer.allocate(remain)
             while (remain > 0) {
                 val read = channel.read(buffer)
-                if (read > 0)
-                    remain = read - read.toInt
+                if (read >= 0) {
+                    remain -= read
+                } else {
+                    // read < 0 means premature end of file => terminate loop
+                    remain = 0
+                }
```

Comment:
I posted and deleted two comments this morning as I try to remember what my concern was previously when I reviewed this PR. I can neither find that comment or remember exactly what it was but there are two high level bits. One, same as Markus' - test for the corner case? Second, looking back at the code, `read()` should not return -1 since we are reading a pre-determined number of bytes `pool.getDockerLogContent(con.containerId, con.lastLogSize, size, runningInContainer)`.